OSGI Missing Provide-Capability entry for liquibase.logging.mdc.MdcManager

### Search first

- [X] I searched and no similar issues were found

### Description

OSGI needs some special manifest entries for service loaders. This problem was solved via PR https://github.com/liquibase/liquibase/pull/2081. But now I can see similar problem for liquibase.logging.mdc.MdcManager. During execution of scripts I can see this exception:
`21:12:45.497 ERROR [Karaf local console user karaf] Exception caught while executing command
java.lang.RuntimeException: drop
	at tech.pantheon.chronos.liquibase.LiquibaseDrop.performDrop(LiquibaseDrop.java:35) ~[?:?]
	at tech.pantheon.chronos.dev.command.LiquibaseDropAllCommand.execute(LiquibaseDropAllCommand.java:34) ~[?:?]
	at org.apache.karaf.shell.impl.action.command.ActionCommand.execute(ActionCommand.java:84) ~[?:?]
	at org.apache.karaf.shell.impl.console.osgi.secured.SecuredCommand.execute(SecuredCommand.java:68) ~[?:?]
	at org.apache.karaf.shell.impl.console.osgi.secured.SecuredCommand.execute(SecuredCommand.java:86) ~[?:?]
	at org.apache.felix.gogo.runtime.Closure.executeCmd(Closure.java:599) ~[?:?]
	at org.apache.felix.gogo.runtime.Closure.executeStatement(Closure.java:526) ~[?:?]
	at org.apache.felix.gogo.runtime.Closure.execute(Closure.java:415) ~[?:?]
	at org.apache.felix.gogo.runtime.Pipe.doCall(Pipe.java:416) ~[?:?]
	at org.apache.felix.gogo.runtime.Pipe.call(Pipe.java:229) ~[?:?]
	at org.apache.felix.gogo.runtime.Pipe.call(Pipe.java:59) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: java.lang.NullPointerException: Cannot invoke "liquibase.logging.mdc.MdcManager.put(String, String)" because the return value of "liquibase.Scope.getMdcManager()" is null
	at liquibase.Scope.addMdcValue(Scope.java:416) ~[?:?]
	at liquibase.Scope.addMdcValue(Scope.java:406) ~[?:?]
	at liquibase.command.CommandScope.execute(CommandScope.java:193) ~[?:?]
	at liquibase.Liquibase.dropAll(Liquibase.java:2180) ~[?:?]
	at liquibase.Liquibase.dropAll(Liquibase.java:2159) ~[?:?]
	at tech.pantheon.chronos.liquibase.LiquibaseDrop.performDrop(LiquibaseDrop.java:33) ~[?:?]`



### Steps To Reproduce

1) Open karaf 4.4.3
2) Execute some liquibase scripts

### Actual Behavior

_No response_

### Expected/Desired Behavior

_No response_

### Liquibase Version

4.19.1

### Database Vendor & Version

Postgresql 14

### Liquibase Integration

Karaf OSGI 

### Liquibase Extensions

_No response_

### OS and/or Infrastructure Type/Provider

Fedora 37 64bit Openjdk 17

### Additional Context

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR (Thank you!)

The fix was in liquibase 4.21.0, released on April 12.

I can confirm that when using 4.21.0 the below error message, present in 4.19.1, and 4.20.0, has gone away:
```
Caused by: java.lang.NullPointerException: Cannot invoke "liquibase.logging.mdc.MdcManager.put(String, String)" because the return value of "liquibase.Scope.getMdcManager()" is null
```

Unfortunately 4.21.0 has a new, and more cryptic error message, and fails to create a database schema (the last liquibase version that works for me, is 4.19.0):
```
Error creating sampleapp test database schema
liquibase.exception.LiquibaseException: java.lang.RuntimeException: Cannot end scope hbjkumzpzy when currently at scope pmyynvrxeq
	at liquibase.Liquibase.runInScope(Liquibase.java:1623) ~[?:?]
	at liquibase.Liquibase.update(Liquibase.java:229) ~[?:?]
	at liquibase.Liquibase.update(Liquibase.java:213) ~[?:?]
	at liquibase.Liquibase.update(Liquibase.java:198) ~[?:?]
	at liquibase.Liquibase.update(Liquibase.java:187) ~[?:?]
	at no.priv.bang.karaf.sample.db.liquibase.test.SampleDbLiquibaseRunner.applyLiquibaseChangelist(SampleDbLiquibaseRunner.java:77) ~[?:?]
	at no.priv.bang.karaf.sample.db.liquibase.test.SampleDbLiquibaseRunner.prepare(SampleDbLiquibaseRunner.java:55) ~[?:?]
	at org.ops4j.pax.jdbc.config.impl.DataSourceRegistration.<init>(DataSourceRegistration.java:88) ~[?:?]
	at org.ops4j.pax.jdbc.config.impl.DataSourceConfigManager.lambda$null$4(DataSourceConfigManager.java:95) ~[?:?]
	at org.ops4j.pax.jdbc.config.impl.ServiceTrackerHelper$1.addingService(ServiceTrackerHelper.java:132) ~[?:?]
	at org.osgi.util.tracker.ServiceTracker$Tracked.customizerAdding(ServiceTracker.java:943) ~[osgi.core-8.0.0.jar:?]
	at org.osgi.util.tracker.ServiceTracker$Tracked.customizerAdding(ServiceTracker.java:871) ~[osgi.core-8.0.0.jar:?]
	at org.osgi.util.tracker.AbstractTracked.trackAdding(AbstractTracked.java:256) ~[osgi.core-8.0.0.jar:?]
	at org.osgi.util.tracker.AbstractTracked.track(AbstractTracked.java:229) ~[osgi.core-8.0.0.jar:?]
	at org.osgi.util.tracker.ServiceTracker$Tracked.serviceChanged(ServiceTracker.java:903) ~[osgi.core-8.0.0.jar:?]
	at org.apache.felix.framework.EventDispatcher.invokeServiceListenerCallback(EventDispatcher.java:990) ~[?:?]
	at org.apache.felix.framework.EventDispatcher.fireEventImmediately(EventDispatcher.java:838) ~[?:?]
	at org.apache.felix.framework.EventDispatcher.fireServiceEvent(EventDispatcher.java:545) ~[?:?]
	at org.apache.felix.framework.Felix.fireServiceEvent(Felix.java:4863) ~[?:?]
	at org.apache.felix.framework.Felix.registerService(Felix.java:3834) ~[?:?]
	at org.apache.felix.framework.BundleContextImpl.registerService(BundleContextImpl.java:328) ~[?:?]
	at org.apache.felix.framework.BundleContextImpl.registerService(BundleContextImpl.java:302) ~[?:?]
	at org.ops4j.pax.jdbc.derby.impl.Activator.start(Activator.java:34) ~[?:?]
	at org.apache.felix.framework.util.SecureAction.startActivator(SecureAction.java:849) ~[?:?]
	at org.apache.felix.framework.Felix.activateBundle(Felix.java:2429) ~[?:?]
	at org.apache.felix.framework.Felix.startBundle(Felix.java:2335) ~[?:?]
	at org.apache.felix.framework.BundleImpl.start(BundleImpl.java:1006) ~[?:?]
	at org.apache.felix.framework.BundleImpl.start(BundleImpl.java:992) ~[?:?]
	at org.apache.karaf.features.internal.service.BundleInstallSupportImpl.startBundle(BundleInstallSupportImpl.java:165) ~[?:?]
	at org.apache.karaf.features.internal.service.FeaturesServiceImpl.startBundle(FeaturesServiceImpl.java:1160) ~[?:?]
	at org.apache.karaf.features.internal.service.Deployer.deploy(Deployer.java:1041) ~[?:?]
	at org.apache.karaf.features.internal.service.FeaturesServiceImpl.doProvision(FeaturesServiceImpl.java:1069) ~[?:?]
	at org.apache.karaf.features.internal.service.FeaturesServiceImpl.lambda$doProvisionInThread$13(FeaturesServiceImpl.java:1004) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
	Suppressed: liquibase.exception.UnexpectedLiquibaseException: java.sql.SQLNonTransientConnectionException: No current connection.
		at liquibase.database.jvm.JdbcConnection.getURL(JdbcConnection.java:149) ~[?:?]
		at liquibase.database.core.DerbyDatabase.close(DerbyDatabase.java:158) ~[?:?]
		at liquibase.Liquibase.close(Liquibase.java:1631) ~[?:?]
		at no.priv.bang.karaf.sample.db.liquibase.test.SampleDbLiquibaseRunner.applyLiquibaseChangelist(SampleDbLiquibaseRunner.java:76) ~[?:?]
		at no.priv.bang.karaf.sample.db.liquibase.test.SampleDbLiquibaseRunner.prepare(SampleDbLiquibaseRunner.java:55) ~[?:?]
		at org.ops4j.pax.jdbc.config.impl.DataSourceRegistration.<init>(DataSourceRegistration.java:88) ~[?:?]
		at org.ops4j.pax.jdbc.config.impl.DataSourceConfigManager.lambda$null$4(DataSourceConfigManager.java:95) ~[?:?]
		at org.ops4j.pax.jdbc.config.impl.ServiceTrackerHelper$1.addingService(ServiceTrackerHelper.java:132) ~[?:?]
		at org.osgi.util.tracker.ServiceTracker$Tracked.customizerAdding(ServiceTracker.java:943) ~[osgi.core-8.0.0.jar:?]
		at org.osgi.util.tracker.ServiceTracker$Tracked.customizerAdding(ServiceTracker.java:871) ~[osgi.core-8.0.0.jar:?]
		at org.osgi.util.tracker.AbstractTracked.trackAdding(AbstractTracked.java:256) ~[osgi.core-8.0.0.jar:?]
		at org.osgi.util.tracker.AbstractTracked.track(AbstractTracked.java:229) ~[osgi.core-8.0.0.jar:?]
		at org.osgi.util.tracker.ServiceTracker$Tracked.serviceChanged(ServiceTracker.java:903) ~[osgi.core-8.0.0.jar:?]
		at org.apache.felix.framework.EventDispatcher.invokeServiceListenerCallback(EventDispatcher.java:990) ~[?:?]
		at org.apache.felix.framework.EventDispatcher.fireEventImmediately(EventDispatcher.java:838) ~[?:?]
		at org.apache.felix.framework.EventDispatcher.fireServiceEvent(EventDispatcher.java:545) ~[?:?]
		at org.apache.felix.framework.Felix.fireServiceEvent(Felix.java:4863) ~[?:?]
		at org.apache.felix.framework.Felix.registerService(Felix.java:3834) ~[?:?]
		at org.apache.felix.framework.BundleContextImpl.registerService(BundleContextImpl.java:328) ~[?:?]
		at org.apache.felix.framework.BundleContextImpl.registerService(BundleContextImpl.java:302) ~[?:?]
		at org.ops4j.pax.jdbc.derby.impl.Activator.start(Activator.java:34) ~[?:?]
		at org.apache.felix.framework.util.SecureAction.startActivator(SecureAction.java:849) ~[?:?]
		at org.apache.felix.framework.Felix.activateBundle(Felix.java:2429) ~[?:?]
		at org.apache.felix.framework.Felix.startBundle(Felix.java:2335) ~[?:?]
		at org.apache.felix.framework.BundleImpl.start(BundleImpl.java:1006) ~[?:?]
		at org.apache.felix.framework.BundleImpl.start(BundleImpl.java:992) ~[?:?]
		at org.apache.karaf.features.internal.service.BundleInstallSupportImpl.startBundle(BundleInstallSupportImpl.java:165) ~[?:?]
		at org.apache.karaf.features.internal.service.FeaturesServiceImpl.startBundle(FeaturesServiceImpl.java:1160) ~[?:?]
		at org.apache.karaf.features.internal.service.Deployer.deploy(Deployer.java:1041) ~[?:?]
		at org.apache.karaf.features.internal.service.FeaturesServiceImpl.doProvision(FeaturesServiceImpl.java:1069) ~[?:?]
		at org.apache.karaf.features.internal.service.FeaturesServiceImpl.lambda$doProvisionInThread$13(FeaturesServiceImpl.java:1004) ~[?:?]
		at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
		at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
		at java.lang.Thread.run(Thread.java:833) ~[?:?]
	Caused by: java.sql.SQLNonTransientConnectionException: No current connection.
		at org.apache.derby.impl.jdbc.SQLExceptionFactory.getSQLException(Unknown Source) ~[?:?]
		at org.apache.derby.impl.jdbc.SQLExceptionFactory.getSQLException(Unknown Source) ~[?:?]
		at org.apache.derby.impl.jdbc.Util.generateCsSQLException(Unknown Source) ~[?:?]
		at org.apache.derby.impl.jdbc.Util.generateCsSQLException(Unknown Source) ~[?:?]
		at org.apache.derby.impl.jdbc.Util.noCurrentConnection(Unknown Source) ~[?:?]
		at org.apache.derby.impl.jdbc.EmbedConnection.checkIfClosed(Unknown Source) ~[?:?]
		at org.apache.derby.impl.jdbc.EmbedConnection.getMetaData(Unknown Source) ~[?:?]
		at liquibase.database.jvm.JdbcConnection.getConnectionUrl(JdbcConnection.java:229) ~[?:?]
		at liquibase.database.jvm.JdbcConnection.getURL(JdbcConnection.java:145) ~[?:?]
		... 34 more
	Caused by: org.apache.derby.iapi.error.StandardException: No current connection.
		at org.apache.derby.iapi.error.StandardException.newException(Unknown Source) ~[?:?]
		at org.apache.derby.impl.jdbc.SQLExceptionFactory.wrapArgsForTransportAcrossDRDA(Unknown Source) ~[?:?]
		at org.apache.derby.impl.jdbc.SQLExceptionFactory.getSQLException(Unknown Source) ~[?:?]
		at org.apache.derby.impl.jdbc.SQLExceptionFactory.getSQLException(Unknown Source) ~[?:?]
		at org.apache.derby.impl.jdbc.Util.generateCsSQLException(Unknown Source) ~[?:?]
		at org.apache.derby.impl.jdbc.Util.generateCsSQLException(Unknown Source) ~[?:?]
		at org.apache.derby.impl.jdbc.Util.noCurrentConnection(Unknown Source) ~[?:?]
		at org.apache.derby.impl.jdbc.EmbedConnection.checkIfClosed(Unknown Source) ~[?:?]
		at org.apache.derby.impl.jdbc.EmbedConnection.getMetaData(Unknown Source) ~[?:?]
		at liquibase.database.jvm.JdbcConnection.getConnectionUrl(JdbcConnection.java:229) ~[?:?]
		at liquibase.database.jvm.JdbcConnection.getURL(JdbcConnection.java:145) ~[?:?]
		... 34 more
Caused by: java.lang.RuntimeException: Cannot end scope hbjkumzpzy when currently at scope pmyynvrxeq
	at liquibase.Scope.exit(Scope.java:241) ~[?:?]
	at liquibase.Scope.child(Scope.java:205) ~[?:?]
	at liquibase.Scope.child(Scope.java:193) ~[?:?]
	at liquibase.Scope.child(Scope.java:172) ~[?:?]
	at liquibase.Liquibase.runInScope(Liquibase.java:1618) ~[?:?]
	... 36 more
```

--- a/liquibase-standard/pom.xml
+++ b/liquibase-standard/pom.xml
@@ -253,7 +253,9 @@
                             osgi.serviceloader; osgi.serviceloader=liquibase.sqlgenerator.SqlGenerator,
                             osgi.serviceloader; osgi.serviceloader=liquibase.logging.mdc.MdcManager,
                             osgi.serviceloader; osgi.serviceloader=liquibase.logging.mdc.CustomMdcObject,
-                            osgi.serviceloader; osgi.serviceloader=liquibase.resource.PathHandler
+                            osgi.serviceloader; osgi.serviceloader=liquibase.resource.PathHandler,
+                            osgi.serviceloader; osgi.serviceloader=liquibase.report.ShowSummaryGenerator,
+                            osgi.serviceloader; osgi.serviceloader=liquibase.parser.LiquibaseSqlParser
                         </Provide-Capability>
                         <Require-Capability>
                             osgi.extender; filter:="(osgi.extender=osgi.serviceloader.registrar)",
@@ -287,7 +289,9 @@
                             osgi.serviceloader; filter:="(osgi.serviceloader=liquibase.sqlgenerator.SqlGenerator)"; cardinality:=multiple,
                             osgi.serviceloader; filter:="(osgi.serviceloader=liquibase.logging.mdc.MdcManager)"; cardinality:=multiple,
                             osgi.serviceloader; filter:="(osgi.serviceloader=liquibase.logging.mdc.CustomMdcObject)"; cardinality:=multiple,
-                            osgi.serviceloader; filter:="(osgi.serviceloader=liquibase.resource.PathHandler)"; cardinality:=multiple
+                            osgi.serviceloader; filter:="(osgi.serviceloader=liquibase.resource.PathHandler)"; cardinality:=multiple,
+                            osgi.serviceloader; filter:="(osgi.serviceloader=liquibase.report.ShowSummaryGenerator)"; cardinality:=multiple,
+                            osgi.serviceloader; filter:="(osgi.serviceloader=liquibase.parser.LiquibaseSqlParser)"; cardinality:=multiple
                         </Require-Capability>
                     </instructions>
                 </configuration>

